src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -665,12 +665,13 @@ private ProcessingBranch PrepareProcessingBranch(Block suggestedBlock, Processin
             // otherwise some nodes would be missing
             bool notFoundTheBranchingPointYet = !_blockTree.IsMainChain(branchingPoint.Hash!);
             bool notReachedTheReorgBoundary = branchingPoint.Number > (_blockTree.Head?.Header.Number ?? 0);
-            preMergeFinishBranchingCondition = (notFoundTheBranchingPointYet || notReachedTheReorgBoundary);
+            bool notInBlockProduction = !options.ContainsFlag(ProcessingOptions.ProducingBlock);
```

Comment:
notInForceProcessing?

Reply:
Can this code be now removed:

https://github.com/NethermindEth/nethermind/pull/6212/files#diff-d3815f34e7dae0c4c0af79174d6c0d34c4c9d5c9e7259c23a56379cea9f72d51R534-R538

Reply:
> notInForceProcessing?

thx, changed.

oh yeah. This code looks unnecessary now, but I am not going to touch it in this PR. Let's analyze it more